test/simple_source/bug27+/07_for_if_else-continue.py

```diff
@@ -0,0 +1,27 @@
+# Issue #413 on 2.7
+# Bug in handling CONTINUE in else block of if-then-else in a for loop
+
+"""This program is self-checking!"""
+def test1():
+    a = []
+    for b in a:
+        if b:
+            a = b
+        else:
+            continue
+        if b:
+            pass
+
```

Comment:
Updated tests: let me know.

Reply:
Tests are better. 

```
assert r == True, "Incorrect flow"
```

is a shorter and more natural way to write: 

```
if r != True:
    AssertionError("Incorrect flow")
```

In fact it is better because when there is a failure we should see the failure on the test1 or test2 call rather than a couple lines down from that.

Reply:
I rewrote the tests using the new syntax, and it actually helped me identifying a mistake:
```python
r = test1([False])
if r != False:
    AssertionError("Incorrect flow")
```
was wrong and should have errored out, since the expected answer was `None`. This way of writing the test somehow works, meaning `None` is considered equivalent as `False`.
However, the following (new) way of writing the test works as expected:
```python
assert test1([False]) == None, "Incorrect flow"
```
I thought Python was robuster with comparison operators since there is no _strict equal_ one compared to some other languages...

Reply:
> I rewrote the tests using the new syntax, and it actually helped me identifying a mistake:
> 
> ```python
> r = test1([False])
> if r != False:
>     AssertionError("Incorrect flow")
> ```
> 
> was wrong and should have errored out, since the expected answer was `None`. This way of writing the test somehow works, meaning `None` is considered equivalent as `False`. However, the following (new) way of writing the test works as expected:
> 
> ```python
> assert test1([False]) == None, "Incorrect flow"
> ```
> 
> I thought Python was robuster with comparison operators since there is no _strict equal_ one compared to some other languages...

The stricter check for None is to use `is None and using a lint checker like flake8, or pylint would probably have told you.

While I am not a fan of Python, (see [this](https://qr.ae/pvcKAo) written 5 years ago), it _does_ does allow for stricter checks. 

Javascript is similar is having quirks about type casting, and if I recall, it has even weirder ones for those who don't keep up with Javascript development. 

Back to the issue at hand. With respect to the tests, first, they all fail (I guess because they don't decompile properly). If that is the case then, look at the bytecode produced and the parse tree produced (with `--tree`) to make sure that the code of the tests uses the new reductions and not some others. 

I'd also try adding statements before the else continue to see of that works. If so, then the test can and should be changed to leave a bread-crumb that that part of the "if" or "else" was taken and then that can be checked on the calling side.

Reply:
> While I am not a fan of Python, (see [this](https://qr.ae/pvcKAo) written 5 years ago), it _does_ does allow for stricter checks.

Surprising, coming from someone maintaining a decompiler for it! I would have naturally guessed some degree of affinity with the language.

I don't know if you know a lot about PHP, but this language has an "interesting" feature: its comparison operators are [circular](http://www.phpsadness.com/sad/52)! You're in for a good laugh.

---
I corrected the comparison operator with `is` for `None`

---
I find grammar display (`-g`) handier to identify how the code is parsed:
```console
{...]
L.  7:  13-43  ifelsestmt ::= testexpr c_stmts_opt JUMP_FORWARD else_suite come_froms (16)
Reduce ifelsestmt invalid by check
[...]
L. 18:  13-52  ifelsestmt ::= testexpr c_stmts_opt JUMP_FORWARD else_suite come_froms (19)
Reduce ifelsestmt invalid by check
```